### PR TITLE
Add Jupyterlab GridStack

### PR DESCRIPTION
**Pull request by jtpio**
Add the `jupyterlab-gridstack` extension from https://github.com/hbcarlos/voila-editor

## TODO

- [x] Bring the original commits
- [x] Change `outputDir` to `voila-gridstack/labextension`
- [x] Check `setup.py` is still correct after the switch to `jupyter-packaging`
- [x] Add tests on CI to check both the classic and lab extensions are correctly installed
- [x] Keep the symlink of template data files?
- [x] Rename lab extension to `@voila-dashboards/jupyterlab-gridstack`
- [x] ~Update to the latest `jupyter_server.pytest_plugin`~ -> tracked in https://github.com/voila-dashboards/voila-gridstack/issues/70

### CODE REVIEW

**Comment by jtpio** [comment]
How about naming the JupyterLab Extension `@voila-dashboards/jupyterlab-gridstack`?

We know have the `voila-dashboard` organization on npm: 

image

Similarly, we could rename `@jupyter-voila/jupyterlab-preview` to `@voila-dashboards/jupyterlab-preview` to stay consistent, and also so it matches the name of the org on GitHub.

File: setup.py [hunk 1]
```diff
@@ -40,56 +47,89 @@ def run(self):
...
+    install_requires=[
+        "jupyterlab_widgets>=1.0.0a6",
+        "voila>=0.2.0,<0.3.0"
     ],
-    'extras_require': {
-        'test': [
+    extras_require={
+        "test": [
             'ipykernel',
+            'jupyter_server~=1.0.1',
```
Comment (jtpio) [inline setup.py:110]: Sticking to this for now since a recent version of `jupyter_server` changed the name of the pytest fixtures.

**Comment by jtpio** [comment]
After a fresh install in a new environment:

```bash
$ jupyter labextension list
JupyterLab v3.0.0rc13
/home/jtp/miniforge3/envs/test/share/jupyter/labextensions
        @jupyter-widgets/jupyterlab-manager v3.0.0-alpha.2 enabled OK
        @voila-dashboards/jupyterlab-gridstack v0.1.0 enabled OK (python, voila-gridstack)
```

> @voila-dashboards/jupyterlab-gridstack v0.1.0 enabled OK (python, voila-gridstack)

This comes from the `install.json` file that tells the user the `@voila-dashboards/jupyterlab-gridstack` extension was installed via the `voila-gridstack` python package.

**Comment by jtpio** [comment]
cc @hbcarlos @SylvainCorlay if you want to check it out.

This should be ready to go. `jupyterlab_widgets` is still in the list of `install_requires` (but not `jupyterlab`), so support for widgets in the editor works out of the box. This should also prevent the issue of a missing `sharedPackages` dependency if `jupyterlab_widgets` is not installed, but specified here:

https://github.com/voila-dashboards/voila-gridstack/blob/27a8402a2a5d9b678034af345615fd75c08d80df/packages/jupyterlab-gridstack/package.json#L74-L83

Just opened https://github.com/voila-dashboards/voila-gridstack/issues/70 to track the remaining item:

> Update to the latest jupyter_server.pytest_plugin

**Comment by hbcarlos** [comment]
Just tried and works perfectly. Thanks Jeremy!

**Comment by jtpio** [comment]
The last commit updates to the latest JupyterLab 3.0 final packages.

We should be able to merge this PR, and proceed with a release soon.

**Comment by jtpio** [comment]
> We should be able to merge this PR, and proceed with a release soon.

Opened https://github.com/voila-dashboards/voila-gridstack/issues/72 to track this.

**Pull request merged**